frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,553 @@
+import { createMachine, assign } from "xstate";
+import { GetMemberListResponse } from "../../utils/fetcher";
+
+type Billing = {
+  id: string;
+  title: string;
+  amount: number;
+};
+
+type BillingDetailMember = {
+  id: string;
+  name: string;
+  amount: number;
+};
+
+type BillingDetail = {
+  id: string;
+  name: string;
+  billAmount: number;
+  chargedMemberId: string;
+  isBillEqually: boolean;
+  members: BillingDetailMember[];
+};
+
+type BillingForm = {
+  title?: string;
+  total?: number;
+  chargedMember?: string;
+  isBillEqual: boolean;
+  members?: GetMemberListResponse["data"];
+};
+
+type FORMMODE = "CREATE" | "EDIT";
+
+interface Context {
+  billings?: Billing[];
+  billingError?: string;
+  formMode?: FORMMODE;
+  members?: GetMemberListResponse["data"];
+  membersError?: string;
+  billingDetail?: BillingDetail;
+  billingDetailError?: string;
+  billingForm?: BillingForm;
+  submitBillingError?: string;
```

Comment:
Yang ini diriku propose pake `null` aja ya instead of `undefined`, alasannya sama kayak yang di atas

Reply:
Yupp yupp